### PR TITLE
Basic AI chat modal

### PR DESCRIPTION
**Pull request by rohan-kadkol**
To speed up the process with the entire AI search feature (and as per @FlyersPh9's suggestion), I will create smaller PRs from time to time. These PRs will be to the `search-docs` feature branch. When the entire feature is ready, we can merge the `search-docs` branch with `main`/`dev`.

This first PR is just with a very basic UI. One of the next PRs will improve this UI, and also make it reactive to the screen size.

[Video30.webm](https://github.com/iTwin/iTwinUI/assets/45748283/42cac046-f0be-47bc-91b0-fcb5f3bd1b66)

## Changes

* A new search button in the header of the docs website. Clicking it opens the `SearchModal`
* `SearchModal` is a `Modal` that has an `Input` and also a `<ReactMarkdown>` to display the answer to the question.

## Testing

Basic manual testing of whether the `SearchModal` opens when the search button in the header of the website is clicked.

## Docs

N/A

### CODE REVIEW

File: apps/website/src/components/Header.astro [hunk 1]
```diff
@@ -45,9 +46,32 @@ const { pathname } = Astro.url;
...
     </a>
+
+    <button id='search-action-item' class='action-item'>
+      <VisuallyHidden>Search</VisuallyHidden>
+
+      <svg width='32' height='32' viewBox='0 0 24 24' fill='currentColor'>
+        <path
+          d='M11 9.7c.7-1 1.1-2.2 1.1-3.5.1-3.5-2.7-6.2-6-6.2C2.7 0 0 2.7 0 6.1s2.7 6.1 6.1 6.1c1.3 0 2.5-.4 3.5-1.1l4.9 4.9 1.4-1.4zm-5 .5c-2.3 0-4.1-1.8-4.1-4.1S3.7 2 6 2s4.1 1.8 4.1 4.1-1.8 4.1-4.1 4.1'
+        ></path>
+      </svg>
```
Comment (rohan-kadkol) [inline apps/website/src/components/Header.astro:57]: I was having a bit of trouble centering the search icon (https://github.com/iTwin/iTwinUI-icons/blob/main/packages/icons/icons/search.svg?short_path=dc61d25). Could someone please take a look at this?

**Comment by mayank99** [comment]
two suggestions:
- The chatbot is not equivalent to search, so it would be better not to use the search icon. Instead it could be a text button that says something like "AI chat".
- We try to avoid using iTwinUI components on the website because it has its own design language, so a custom dialog should be created instead. I can help you with the CSS if you want to pair sometime.

**Comment by rohan-kadkol** [comment]
> two suggestions:
> 
> * The chatbot is not equivalent to search, so it would be better not to use the search icon. Instead it could be a text button that says something like "AI chat".

Sounds good, will make this change.

> * We try to avoid using iTwinUI components on the website because it has its own design language, so a custom dialog should be created instead. I can help you with the CSS if you want to pair sometime.

Thanks! Yes, would be great if you can guide me in the right direction, if you have time.

**Review by gretanausedaite — APPROVED** [review]
Basic UI looks good to me 🌵

**Review by adamhock — APPROVED** [review]
I'm excited to see this!

**Comment by mayank99** [comment]
Going to merge this into `search-docs`.

Thanks @rohan-kadkol. Looking forward to your next PR.

**Comment by rohan-kadkol** [comment]
> Going to merge this into `search-docs`.
> 
> Thanks @rohan-kadkol. Looking forward to your next PR.

Thanks for that, and also thanks for your help with the CSS styling!

I hope to have a few more PRs this week.

**Pull request merged**